Team417_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team417_CENTERSTAGE/BaseAutonomous.java

```diff
@@ -0,0 +1,234 @@
+        telemetry.update();
+        initializeHardware();
+        myColorDetection.init();
+
+        telemetry.addData("Init State", "Init Finished");
+
+        // Set last know encoder values
+        lastEncoderFR = FR.getCurrentPosition();
+        lastEncoderFL = FL.getCurrentPosition();
+        lastEncoderBL = BL.getCurrentPosition();
+        lastEncoderBR = BR.getCurrentPosition();
+
+        sleep(5000);
+
+        telemetry.clear();
+        telemetry.addLine("Initialized. Ready to start!");
+        telemetry.update();
+    }
+
+    private void stopDriving() {
+        FL.setPower(0.0);
+        FR.setPower(0.0);
+        BL.setPower(0.0);
+        BR.setPower(0.0);
+    }
+
+    public void runSimpleInchesAuto(boolean red, boolean close) {
+
+        if (red) {
+            myColorDetection.setDetectColor(OpenCvColorDetection.detectColorType.RED);
+            telemetry.addLine("Looking for red");
+        } else {
+            myColorDetection.setDetectColor(OpenCvColorDetection.detectColorType.BLUE);
+            telemetry.addLine("Looking for blue");
+        }
+
+        initializeAuto();
+
+        waitForStart();
+        Log.d("skid", toString());
+
+        double x = 0;
+        double y = 0;
+        switch (myColorDetection.detectTeamProp()) {
+            case LEFT:
+                telemetry.addData("Side", "Left");
+                x = LEFT_X;
+                y = LEFT_Y;
+                break;
+            case CENTER:
+                telemetry.addData("Side", "Center");
+                x = CENTER_X;
+                y = CENTER_Y;
+                break;
+            case RIGHT:
+                telemetry.addData("Side", "Right");
+                x = RIGHT_X;
+                y = RIGHT_Y;
+                break;
+            default:
+                telemetry.addData("Side", "Unsure");
+        }
+        telemetry.update();
+        driveInches(0, y);
+        driveInches(x, 0);
+
+        if (intakeMotor != null) {
+            intakeMotor.setPower(INTAKE_SPEED);
+            sleep((long) INTAKE_TIME);
+            intakeMotor.setPower(0);
+        } else {
+            sleep(5000);
+        }
+
+        driveInches(-x, 0);
+        driveInches(0, -y);
+        driveInches(0, MOVING_FROM_WALL);
+
+        if (close) {
+            if (red) {
+                driveInches(CLOSE_PARKING, Y_CALIBRATION_RIGHT);
+            } else {
+                driveInches(-CLOSE_PARKING, Y_CALIBRATION_LEFT);
+            }
+        } else {
+            if (red) {
+                driveInches(FAR_PARKING, Y_CALIBRATION_RIGHT);
+            } else {
+                driveInches(-FAR_PARKING, Y_CALIBRATION_LEFT);
+            }
+        }
+
+        if (intakeMotor != null) {
+            intakeMotor.setPower(-INTAKE_SPEED2);
+            sleep((long) INTAKE_TIME2);
+            intakeMotor.setPower(0);
+        } else {
+            sleep(5000);
+        }
+    }
+
+    @Override
```

Comment:
This snippet was auto-generated using Android Studio. A comment would be helpful to explain what it's useful for (for dumping the current settings to a log) and how it was generated (right-click, "Generate...", "toString()" with "Settings..." set to include static and constant fields).

Reply:
Done 👍